app/src/main/java/com/paddlesandbugs/dahdidahdit/brasspound/AbstractPaddleInputActivity.java

```diff
@@ -108,6 +110,17 @@ public boolean dispatchKeyEvent(KeyEvent event) {
         return super.dispatchKeyEvent(event);
     }
 
+    @Override
+    public boolean dispatchTouchEvent(MotionEvent event) {
+        if (event.getSource() != InputDevice.SOURCE_TOUCHSCREEN) {
```

Comment:
Just came to my mind: How about event.getSource() == InputDevice.MOUSE? Would that work? Seems more specific and might solve the other problem, too.

Reply:
Unfortunately with different devices I tried I get different sources (I use a cheap aliexpress vband usb dongle and a real mouse). Also this wouldn't solve the emulator issue. In the emulator my trackpad becomes a trackpad source when hw forwarding is enabled. No way to distinguish this from a screen tap anymore.